### ISSUE TITLE
FEATURE: Native support for a `Neos.Neos:Folder` nodeType without uripathsegment

### ISSUE THREAD

**Issue by mhsdesign** [neos/neos-development-collection#4829]
Naturally every page has a uripathsegment. There is only one exception: the homepage (Neos.Neos:Site) itself.
A new mixin `Neos.Neos:UriPathSegment` https://github.com/neos/neos-development-collection/issues/4621 will be introduced to disable the property there. Now one might come to the conclusion that this could also work out of the box for other nodetypes, but this is probably mistaken, while it would be cool ^^

```yaml
'Neos.Neos:Folder':
  abstract: true
  superTypes:
    'Neos.Neos:Document': true
    'Neos.Neos:UriPathSegment': false
```

I dont know exactly where this will fail as i havent thought nor experimented to heavy with this idea, but i think @bwaidelich already has some information.

Nevertheless this is just an idea and not anticipated for Neos 9.

**Comment by bwaidelich** [comment]
I don't have more information, but I fear that this won't work out of the box because the Neos UI will try to load the preview of a node when it is selected in the tree.. But we can try and find out once the mixin exists :)

**Comment by lorenzulrich** [comment]
Yes, it would be nice to have the functionality of https://github.com/breadlesscode/neos-nodetypes-folder in the core.

**Comment by kitsunet** [comment]
Yeah well, I don'T like this implementation. IMHO there should be distinct lines of inheritance,

DocumentWithUriPathSegment and DocumentWithoutUriPathSegment (names TBD), and then you either inherit from one or the other, no _unset_ of uriPathSegment in some way.

I also still think it wouldn't be too bad to do:

Neos.Neos:Document
- Neos.Neos:DocumentWithUri
- - My.Package:Page
- Neos.Neos:Site
- Neos.Neos.Folder

Although inheriting Document and now having to switch to DocumentWithUri would probably be more oftne the case than if we changed the top level element and you would have to chnage "queries" that you want to include all kinds of documents, so:

Neos.Neos:Folder
- Neos.Neos:Document (which implicitly then has Uri)
- - My.Package:Page
- Neos.Neos:Site
